src/presentation/controllers/device-controller.ts

```diff
@@ -101,4 +101,17 @@ export class DeviceController extends BaseController {
       this.httpErrorHandler.handleError(response, error as Exception)
     }
   }
+
+  @PostRequest('/:deviceId/reconnect')
+  public reconnectDevice(request: Request, response: Response): void {
+    try {
+      const deviceId: string = request.params.deviceId
+
+      this.deviceService.reconnect(deviceId)
+      
+      response.send(this.httpResponseFormatter.formatSuccessResponse({ message: 'Device reconnected successfully.' }))
```

Comment:
I guess this is somewhat a asynchronous vs synchronous/ monolithic vs distributed mindset, but at this moment, when we respond to the client, can guarantee that the device has reconnected successfully?